### PR TITLE
Add config file for secret values

### PR DESCRIPTION
**Pull request by eternaltyro**
- Add configuration file into which secrets and credentials can be added
  in order to avoid exposing passwords to the terminal
- Modify connection parameters for database to fallback to config file values
- Order of precedence is passed arguments -> ENVVARS -> config file values
- Modify Argparser help strings and defaults
- Add exception handling for psycopg2 connections

### CODE REVIEW

**Review by omranlm — CHANGES_REQUESTED** [review]
Please check the HATABASE_HOST env when picking ENVVAR om config.py

File: config.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# Type defaults here
+_DATABASE_HOST = None
+_DATABASE_PORT = '5432'
+_DATABASE_NAME = 'postgres'
+_DATABASE_USER = 'postgres_dba'
+_DATABASE_PASSWORD = None
+
+# Pick from ENVVAR if available, else fallback to values above
+DATABASE_HOST = os.getenv('DATABASE_PASSWORD', _DATABASE_HOST)
```
[the remark this answers is not in the document]
Comment (eternaltyro) [reply config.py]: Nice catch. delivered a fix. @omranlm

**Comment by omranlm** [comment]
Fixed another type for frequency and committed change

**Comment by eternaltyro** [comment]
Thanks for the fix and merge @omranlm

**Pull request merged**